aiida_common_workflows/workflows/relax/nwchem/generator.py

```diff
@@ -0,0 +1,143 @@
+        threshold_stress: float = None,
+        previous_workchain=None,
+        **kwargs
+    ) -> engine.ProcessBuilder:
+        """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
+
+        :param structure: the structure to be relaxed.
+        :param calc_engines: a dictionary containing the computational resources for the relaxation.
+        :param protocol: the protocol to use when determining the workchain inputs.
+        :param relax_type: the type of relaxation to perform.
+        :param electronic_type: the electronic character that is to be used for the structure.
+        :param spin_type: the spin polarization type to use for the calculation.
+        :param magnetization_per_site: a list with the initial spin polarization for each site. Float or integer in
+            units of electrons. If not defined, the builder will automatically define the initial magnetization if and
+            only if `spin_type != SpinType.NONE`.
+        :param threshold_forces: target threshold for the forces in eV/Å.
+        :param threshold_stress: target threshold for the stress in eV/Å^3.
+        :param previous_workchain: a <Code>RelaxWorkChain node.
+        :param kwargs: any inputs that are specific to the plugin.
+        :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
+        """
+        # pylint: disable=too-many-locals
+        protocol = protocol or self.get_default_protocol_name()
+
+        super().get_builder(
+            structure,
+            calc_engines,
+            protocol=protocol,
+            relax_type=relax_type,
+            electronic_type=electronic_type,
+            spin_type=spin_type,
+            magnetization_per_site=magnetization_per_site,
+            threshold_forces=threshold_forces,
+            threshold_stress=threshold_stress,
+            previous_workchain=previous_workchain,
+            **kwargs
+        )
+
+        # Protocol
+        parameters = self.get_protocol(protocol)
+        _ = parameters.pop('description')
+
+        # Relaxation type
+        if relax_type == RelaxType.ATOMS:
+            parameters['task'] = 'band optimize'
+        elif relax_type == RelaxType.ATOMS_CELL:
+            parameters['task'] = 'band optimize'
+            parameters['set'] = {'includestress': '.true.'}
+        else:
+            raise ValueError('relax_type `{}` is not supported'.format(relax_type.value))
+
+        # Electronic type
+        if electronic_type == ElectronicType.INSULATOR:
+            pass
+        elif electronic_type == ElectronicType.METAL:
+            parameters['nwpw'] = {'smear': 'fermi', 'scf': 'Anderson outer_iterations 0 Kerker 2.0'}
+        else:
+            raise ValueError('electronic_type `{}` is not supported'.format(electronic_type.value))
+
+        # Spin type
+        if spin_type == SpinType.NONE:
+            pass
+        if spin_type == SpinType.COLLINEAR:
```

Comment:
This should be `elif` otherwise it will raise for `SpinType.NONE`.

Reply:
Also if you actually support `SpinType.COLLINEAR` you should add it to the `_spin_types` class attribute